src/ray/core_worker/core_worker.cc

```diff
@@ -1176,12 +1175,12 @@ void CoreWorker::Exit(
     const rpc::WorkerExitType exit_type,
     const std::string &detail,
     const std::shared_ptr<LocalMemoryBuffer> &creation_task_exception_pb_bytes) {
-  if (is_exit_) {
```

Comment:
@kevin85421 I should've caught this in the previous PR review, but this was a race condition bug -- multiple threads could pass this check prior to setting `is_exit_ = true` on the following line.

Reply:
Good catch! I just copied and pasted from Shutdown. I should have thought twice before copying and pasting.